File: 0001-two-sum/0001-two-sum.js
var twoSum = function(nums, target) {
    const tempMap = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        let result = target - nums[i];
        
        if (tempMap.has(result)) {
            return [i, tempMap.get(result)];
        }
        
        tempMap.set(nums[i], i);
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) { // Time: O(n), Space: O(1)

        for (int i = 1; i < nums.length; i++) {

            for (int j = i; j < nums.length; j++) {

                if (nums[j] + nums[j - i] == target) {
                    return new int[] { j - i, j };
                }
            }
        }

        return new int[] {};
    }
}